tests/test_execution.py

```diff
@@ -5,22 +5,14 @@
 import substratest as sbt
 
 
-def test_tuples_execution_on_same_node(factory, session):
+def test_tuples_execution_on_same_node(data_network):
     """Execution of a traintuple, a following testtuple and a following traintuple."""
-    spec = factory.create_dataset()
-    dataset = session.add_dataset(spec)
+    factory, network = data_network
+    session = network.sessions[0]
```

Comment:
we should do a copy of the session here (to avoid updating the state of the data network session) ?

Reply:
Good point, it's less risky that way!